onnxruntime/core/optimizer/gelu_fusion.cc

```diff
@@ -0,0 +1,168 @@
+    ORT_RETURN_IF_ERROR(Recurse(node, modified, graph_level));
+
+    if (!graph_utils::IsSupportedOptypeVersionAndDomain(node, "Div", {7}) ||
+        !graph_utils::IsSupportedProvider(node, GetCompatibleExecutionProviders()) ||
+        node.GetOutputEdgesCount() != 1) {
+      continue;
+    }
+    std::tuple<bool, NodeArg*, NodeArg*> t;
+    IsInputConstant(graph, node, t);
+    if (!std::get<0>(t) || std::get<1>(t) == nullptr || std::get<2>(t) == nullptr) {
+      continue;
+    }
+
+    auto erf_node_itr = node.OutputNodesBegin();
+    if (erf_node_itr == node.OutputNodesEnd()) {
+      continue;
+    }
+
+    const Node& erf_node = (*erf_node_itr);
+    if (!graph_utils::IsSupportedOptypeVersionAndDomain(erf_node, "Erf", {9}) ||
+        erf_node.GetExecutionProviderType() != node.GetExecutionProviderType() ||
+        erf_node.GetOutputEdgesCount() != 1) {
+      continue;
+    }
+
+    auto add_node_itr = erf_node.OutputNodesBegin();
+    if (add_node_itr == erf_node.OutputNodesEnd()) {
+      continue;
+    }
+
+    const Node& add_node = (*add_node_itr);
+    std::tuple<bool, NodeArg*, NodeArg*> add_input_check;
+    IsInputConstant(graph, add_node, add_input_check);
+    if (!std::get<0>(add_input_check) || std::get<1>(add_input_check) == nullptr || std::get<2>(add_input_check) == nullptr) {
+      continue;
+    }
+    if (!graph_utils::IsSupportedOptypeVersionAndDomain(add_node, "Add", {7}) ||
+        add_node.GetExecutionProviderType() != node.GetExecutionProviderType() ||
+        add_node.GetOutputEdgesCount() != 1) {
+      continue;
+    }
+
+    auto mul_node_itr = add_node.OutputNodesBegin();
+    if (mul_node_itr == add_node.OutputNodesEnd()) {
+      continue;
+    }
+
+    const Node& mul_node = *(mul_node_itr);
+    if (!graph_utils::IsSupportedOptypeVersionAndDomain(mul_node, "Mul", {7}) ||
+        mul_node.GetExecutionProviderType() != node.GetExecutionProviderType()) {
+      continue;
+    }
+
+    const Node* mul2_node = nullptr;
+    for (auto iter = mul_node.InputNodesBegin(); iter != mul_node.InputNodesEnd(); ++iter) {
+      if ((*iter).OpType().compare("Mul") == 0) {
+        // find the other input node of Mul
+        mul2_node = &(*iter);
+        break;
+      }
+    }
+    if (mul2_node == nullptr) {
+      continue;
+    }
+
+    std::tuple<bool, NodeArg*, NodeArg*> mul2_input_check;
+    IsInputConstant(graph, *mul2_node, mul2_input_check);
+    if (!std::get<0>(mul2_input_check) || std::get<1>(mul2_input_check) == nullptr || std::get<2>(mul2_input_check) == nullptr) {
+      continue;
+    }
+    if (!graph_utils::IsSupportedOptypeVersionAndDomain(*mul2_node, "Mul", {7}) ||
+        mul2_node->GetExecutionProviderType() != node.GetExecutionProviderType() ||
+        mul2_node->GetOutputEdgesCount() != 1) {
+      continue;
+    }
+
+    const std::vector<NodeArg*> gelu_input_defs{std::get<1>(t)};
+    const std::vector<NodeArg*> gelu_output_defs{const_cast<NodeArg*>(mul_node.OutputDefs()[0])};
```

Comment:
Does there need to be a check on the actual constant values used in the Div, Add and Mul nodes? I'm assuming specific values are required for Gelu to replace those nodes without this information being an input. e.g. can you fuse Div  with B of 100 the same as Div with B of 2?

Reply:
yeah, this is a todo. I  will fix in next update.